assets/js/containers.js

```diff
@@ -0,0 +1,914 @@
+
```

Comment:
Could you make all the changes to this file in a separate commit so that git sees this as a move and retains the file history. I added some comments for js problems, feel free to ignore them and I can do a follow up fix if you'd like.